app/javascript/components/public_health/query/AdvancedFilter.js

```diff
@@ -757,30 +777,124 @@ class AdvancedFilter extends React.Component {
     );
   };
 
+  renderAdvancedFilterModal = () => {
+    return (
+      <Modal
+        id="advanced-filter-modal"
+        show={this.state.show}
+        centered
+        dialogClassName="modal-af"
+        className="advanced-filter-modal-container"
+        onHide={this.cancel}>
+        <Modal.Header>
+          <Modal.Title>Advanced Filter: {this.state.activeFilter ? this.state.activeFilter.name : 'untitled'}</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <Row className="pb-2 pt-1">
+            <Col>
+              {!this.state.activeFilter && (
+                <Button
+                  id="advanced-filter-save"
+                  variant="primary"
+                  onClick={() => {
+                    this.setState({ showFilterNameModal: true, show: false });
+                  }}
+                  className="mr-1">
+                  <i className="fas fa-save"></i>
+                  <span className="ml-1">Save</span>
+                </Button>
+              )}
+              {this.state.activeFilter && (
+                <Button id="advanced-filter-update" variant="primary" onClick={this.update} className="mr-1">
+                  <i className="fas fa-marker"></i>
+                  <span className="ml-1">Update</span>
+                </Button>
+              )}
+              {this.state.activeFilter && (
+                <Button id="advanced-filter-delete" variant="danger" onClick={this.delete} disabled={!this.state.activeFilter}>
+                  <i className="fas fa-trash"></i>
+                  <span className="ml-1">Delete</span>
+                </Button>
+              )}
+              <div className="float-right">
+                <Button id="advanced-filter-reset" variant="danger" onClick={this.reset}>
```

Comment:
Found a bug (or maybe some undesired behavior) when:

1. Create any kind of advanced filter (I just made one with monitoring = true)
2. Apply the filter
3. Click on "Advanced Filter" to reopen/edit the current filter
4. Click "Reset" then click "Cancel" instead of "Apply"

Notice how the advanced filter was reset but the table results are not updated

Reply:
That is currently how the reset behavior works - it resets the modal but not the table.  I think @kylietmo created a ticket for revisiting how the reset functionality works

Reply:
That I did because I also think it behaves unexpectedly @timwongj. Ticket is here: https://tracker.codev.mitre.org/browse/SARAALERT-1232

Reply:
sounds good, just wanted to make sure we were aware of it